### DISCUSSION TITLE
Deprecation logs

### DISCUSSION THREAD

**Discussion by jasonvarga** [spatie/laravel-ray#246] [Q&A]
Is there a suggested way to have Ray ignore deprecation log messages?

For example on some sites I get flooded with this sort of thing:

![image](https://user-images.githubusercontent.com/105211/161827032-6906e0cf-760d-4419-9d60-37a7e6e17e25.png)

**Comment by freekmurze** [comment]
Are you using the latest version of laravel-ray?

We took care of it here: https://github.com/spatie/laravel-ray/pull/229

**Reply by jasonvarga** [reply]
I'm running these versions:

```
spatie/laravel-ray                 1.29.5   Easily debug Laravel apps
spatie/ray                         1.34.1   Debug with Ray to fix problems faster
```

**Reply by freekmurze** [reply]
@Nielsvanpach could you take a look at this one?

**Reply by jasonvarga** [reply]
Let me know if I can provide any other information that'd help.

**Reply by freekmurze** [reply]
If you have the time then sharing a minimal demo app that demonstrates the problem you're seeing would be handy to debug.

**Reply by jasonvarga** [reply]
I'm having a tough time reproducing it on a fresh site. When I figure it out I'll update this discussion.

**Reply by jasonvarga** [reply]
Here you go. I worked out it's because the `DeprecatedNoticeWatcher` only considers a log message deprecated if it contains the word "deprecated".

https://github.com/jasonvarga/ray-bug

You should probably be checking if a message is deprecated by the level, like Laravel does here:
https://github.com/laravel/framework/blob/2f51de56f1a4d207c3bd82984a96882f9da606fe/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php#L70

The error in my demo repo is considered deprecated by Laravel but not by Ray.

**Reply by Nielsvanpach** [reply]
Hi @jasonvarga, a fix has been tagged in [1.29.6](https://github.com/spatie/laravel-ray/releases/tag/1.29.6).

**Reply by jasonvarga** [reply]
Thanks
